[#11878] Update SearchAccountRequests endpoint

EuniceSim142 opened the pull request:
Part of #11878.

**Outline of solution**
- Add `comments` and `status` as search-able fields for account requests.
- Update `solr.sh` script to generate solr schema with new 'comments' and 'status' fields.

domoberzin reviewed (commented):
Code looks good to me, but should some tests be added to `AccountRequestSearchIT` for these new fields?

EuniceSim142 commented:
IT tests have been added, and updated search action to use id to query db for account request entities.

AccountRequestSearchIndexing action also updated to not include `java.util.UUID:` for id field (Don't think it's necessary and makes converting from solr document field's value to UUID object easier)
| Before: Without `toString()` | Now: with `toString()` |
| -- | -- |
| <img width="495" alt="Screenshot 2024-03-28 at 5 21 37 PM" src="https://github.com/TEAMMATES/teammates/assets/77243938/2a3960ca-0803-4178-baf9-7f16139e8b68"> | <img width="372" alt="Screenshot 2024-03-28 at 5 21 43 PM" src="https://github.com/TEAMMATES/teammates/assets/77243938/f8304ec8-f275-4d96-9c0b-53cc91e00288"> |

jayasting98 reviewed (commented):
It seems okay to me, the failing check just needs to be addressed.

domoberzin commented on src/main/java/teammates/storage/sqlapi/AccountRequestsDb.java:
```diff
@@ -155,7 +155,7 @@ public void deleteDocumentByAccountRequestId(UUID accountRequestId) {
             // Solr saves the id with the prefix "java.util.UUID:", so we need to add it here to
```
I think this comment can be removed since the changes have been made below

domoberzin reviewed (commented):
Other than the one comment below, all good to me

domoberzin reviewed (approved):
LGTM!

ziqing26 reviewed (approved):
@EuniceSim142 @domoberzin Thanks a lot for managing to fix the failing tests! LGTM 🙏
